codegen-server/python/src/main/kotlin/software/amazon/smithy/rust/codegen/server/python/smithy/generators/PythonServerEnumGenerator.kt

```diff
@@ -28,19 +33,29 @@ class PythonServerEnumGenerator(
     runtimeConfig: RuntimeConfig,
 ) : ServerEnumGenerator(model, symbolProvider, writer, shape, enumTrait, runtimeConfig) {
 
+    val pyo3Symbols = listOf(PythonServerCargoDependency.PyO3.asType())
+
     override fun render() {
-        writer.renderPyClass(shape)
+        renderPyClass()
         super.render()
         renderPyO3Methods()
     }
 
+    private fun renderPyClass() {
+        if (shape.hasTrait<ErrorTrait>())
+            Attribute.Custom("pyo3::pyclass(extends = pyo3::exceptions::PyException)", symbols = pyo3Symbols).render(writer)
+        else
+            Attribute.Custom("pyo3::pyclass", symbols = pyo3Symbols).render(writer)
+    }
```

Comment:
FYI a `Enum` cannot have an `ErrorTrait` attached, so this can be just

```kotlin
Attribute.Custom("pyo3::pyclass", symbols = pyo3Symbols).render(writer)
```

Reply:
Oh absolutely right. Have modified the code. Thanks